src/Npgsql/NpgsqlBinaryImporter.cs

```diff
@@ -51,18 +51,19 @@ internal NpgsqlBinaryImporter(NpgsqlConnector connector, string copyFromCommand)
             _buf = connector.WriteBuffer;
             _column = -1;
 
-            try
-            {
-                _connector.SendQuery(copyFromCommand);
+            _connector.SendQuery(copyFromCommand);
 
-                CopyInResponseMessage copyInResponse;
-                var msg = _connector.ReadMessage();
-                switch (msg.Code)
-                {
+            CopyInResponseMessage copyInResponse;
+            var msg = _connector.ReadMessage();
+            switch (msg.Code)
+            {
                 case BackendMessageCode.CopyInResponse:
                     copyInResponse = (CopyInResponseMessage)msg;
                     if (!copyInResponse.IsBinary)
+                    {
+                        _connector.Break();
                         throw new ArgumentException("copyFromCommand triggered a text transfer, only binary is allowed", nameof(copyFromCommand));
+                    }
```

Comment:
nit: This condition could be inverted.

Reply:
Yeah, no strong feelings from me here - all things equal I generally prefer exceptions to be handled in conditions.

Reply:
Yes, that is what I meant. But, it's just a personal preference. If you don't think it helps the clarity here, feel free to disregard.